source/1_engine/rex_engine/src/win/win_com_library.cpp

```diff
@@ -0,0 +1,102 @@
+#include "rex_engine/win/win_com_library.h"
+
+#include <coml2api.h>
+#include <objbase.h>
+#include <objidl.h>
+#include <shlguid.h>
+#include <shobjidl_core.h>
+#include <wtypes.h>
+
+thread_local card32 rex::win::ComLibrary::s_init_succeeded_count = 0;
+
+rex::win::ComLibrary::ComLibrary()
+{
+  if(s_init_succeeded_count == 0)
+  {
+    if(init_lib())
+    {
+      inc_ref();
+    }
+  }
+  else
+  {
+    inc_ref();
+  }
+}
+
+rex::win::ComLibrary::~ComLibrary()
+{
+  if(s_init_succeeded_count == 1)
+  {
+    uninit_lib();
+  }
+
+  dec_ref();
+}
+
+bool rex::win::ComLibrary::is_initialized() const
+{
+  return s_init_succeeded_count > 0;
+}
+
+rsl::string rex::win::ComLibrary::read_link(rsl::string_view filepath)
```

Comment:
Still think this is a bit of redundant code (syslink, junctions, etc.). I don't think it will ever be used =/ 
It's nice that you are able to implement this, but we should not use REX as a show off =P